src/hooks/analytics.jsx

```diff
@@ -0,0 +1,97 @@
+import {
+  createRef, useCallback, useContext, useMemo,
+} from 'react';
+
+import { ParagonContext } from '../ParagonProvider';
+
+/**
+ * Returns the ``analytics`` key stored in ParagonProvider's context
+ */
+export const useAnalytics = () => {
+  const { analytics } = useContext(ParagonContext) || {};
+  return analytics;
+};
+
+/**
+ *
+ * @param {object} [event] Object containing the analytic event name, properties, and dispatchers.
+ * @param {function} [onClick] Custom event handler to call when processing the onClick event.
+ * @param {object} [forwardedRef] An optional React forwarded ref.
+ *
+ * @returns {array} Array containing an onClick event handler to pass to an element that will dispatch
+ *  an analytics event, and an element ref.
+ */
+export const useHandleLogClick = ({
+  event, onClick, forwardedRef,
+}) => {
+  const ref = useMemo(
+    () => forwardedRef || createRef(),
+    [forwardedRef],
+  );
+  const analytics = useAnalytics();
+  const {
+    name: eventName,
+    properties: eventProperties,
+    dispatchers: eventDispatchers,
+  } = event || {};
+
+  /**
+   * Iterates through all analytic event dispatch functions passed in, passing the event name and
+   * properties to each.
+   */
+  const sendEventToDispatchers = useCallback(
+    () => {
+      if (!eventDispatchers) {
+        return;
+      }
+      eventDispatchers.filter(dispatcher => typeof dispatcher === 'function').forEach((dispatch) => {
+        dispatch(eventName, eventProperties);
```

Comment:
TODO: any way to make this more resilient?

Reply:
If an error were to occur when calling the provided `dispatch` function(s), we could either swallow the error here (e.g., with a try/catch) or let it surface the issue to the MFE.

We are opting to let the error occur to make sure it is observable/known and picked up by our tooling like NewRelic.